deeprank/learn/DataSet.py

```diff
@@ -535,36 +524,35 @@ def filter(self, molgrp):
         that must be either of the form: { 'name': cond } or None
 
         Args:
-            molgrp (str): group name of the molecule in the hdf5 file
+            molgrp (hdf5 group): group of the molecule in the hdf5 file
         Returns:
             bool: True if we keep the complex False otherwise
 
         Raises:
             ValueError: If an unsuported condition is provided
         """
+
+        logger.debug("filtering {}".format(molgrp.name))
+
         if self.dict_filter is None:
             return True
 
         for cond_name, cond_vals in self.dict_filter.items():
 
-            try:
-                val = molgrp['targets/' + cond_name][()]
-            except KeyError:
-                warnings.warn(f'Filter {cond_name} not found for mol '
-                              f'{molgrp.name}')
+            logger.debug("filter with condition {}: {}".format(cond_name, cond_vals))
+
```

Comment:
cond_name and cond_vals example- 'Benign':'0.0' or 'Pathogenic': '1.0' ?
If we plan on additional class labels (I'll ask Bas) we can add.

Reply:
I see that right now, the filter function uses the operators '>=', '<=', '==', '>', '<'. So '!=' is missing. Maybe we still need to change it then..

Reply:
Since the values we have are binary, may be <=, >=, >, < may not make sense. 
!= can be added, yes

Reply:
I added !=